performance/objcinterop/src/nativeInterop/cinterop/complexNumbers.m

```diff
@@ -0,0 +1,61 @@
+#import <stdio.h>
+#import "complexNumbers.h"
+
+@interface Complex ()
+- (void)zeroComplex;
+@end
+
+@implementation Complex
+- (id)init {
+    return [self initWithRe: 0.0 andIm: 0.0];
+}
+
+- (id)initWithRe: (double)_re andIm: (double)_im {
+    if (self = [super init]) {
+        re = _re;
+        im = _im;
+        format = @"re: %.1lf im: %.1lf";
+    }
+    return self;
+}
+
++ (Complex *)complexWithRe: (double)re andIm: (double)im {
+    return [[Complex alloc] initWithRe: re andIm: im];
```

Comment:
Do you compile this file with ARC enabled?

Reply:
Yes, as I understood, this is modern way of writting on ObjC and Xcode has such settings by default. Could it work with Kotlin properly? In ObjC I used `@autoreleasepool` in main to test.

Reply:
> Yes

Then `retain` here shouldn't be required:
https://github.com/JetBrains/kotlin-native/pull/2852/files#diff-c4aac30e2bfe14c88545c761df9a7809R13

> Could it work with Kotlin properly?

Yes, it works with Kotlin properly.